editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/bean/AttributeTextFieldPropertyUICreator.java

```diff
@@ -0,0 +1,78 @@
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.properties.bean;
+
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.observable.map.IObservableMap;
+import org.eclipse.jface.databinding.swt.WidgetProperties;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetWidgetFactory;
+import org.fusesource.ide.camel.editor.properties.creators.AbstractTextFieldParameterPropertyUICreator;
+import org.fusesource.ide.camel.editor.properties.creators.modifylisteners.text.AbstractTextParameterPropertyModifyListener;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+import org.fusesource.ide.camel.model.service.core.catalog.eips.Eip;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelBean;
+import org.w3c.dom.Node;
+
+/**
+ * @author brianf
+ *
+ */
+public class AttributeTextFieldPropertyUICreator extends AbstractTextFieldParameterPropertyUICreator {
+
+	public AttributeTextFieldPropertyUICreator(DataBindingContext dbc, IObservableMap<String, String> modelMap, Eip eip, AbstractCamelModelElement camelModelElement, Parameter parameter,
+			Composite parent, TabbedPropertySheetWidgetFactory widgetFactory) {
+		super(dbc, modelMap, eip, camelModelElement, parameter, parent, widgetFactory, 
+				new AttributeTextParameterPropertyModifyListenerForAdvanced(camelModelElement, parameter));
+	}
+
+	public AttributeTextFieldPropertyUICreator(DataBindingContext dbc, IObservableMap<String, String> modelMap, Eip eip, AbstractCamelModelElement camelModelElement, Parameter parameter,
+			Composite parent, TabbedPropertySheetWidgetFactory widgetFactory, AbstractTextParameterPropertyModifyListener listener) {
+		super(dbc, modelMap, eip, camelModelElement, parameter, parent, widgetFactory, listener);
+	}
+
+	@Override
+	public String getInitialValue() {
+		final String parameterName = parameter.getName();
+		final Object parameterValue = camelModelElement.getParameter(parameterName);
+		final Parameter param = eip.getParameter(parameterName);
+		final String defaultValue = param.getDefaultValue();
+		if (camelModelElement instanceof CamelBean && parameterValue == null) {
+			final Object attrValue = getAttributeValue(parameterName);
+			return (String) (attrValue != null ? attrValue : param != null ? defaultValue : "");
```

Comment:
it is the first time that I encounter nested ternary operations and I spent 5 minutes to understand what the code is effectively doing.
Perhaps it was because it was the first time for me and I just need to be accustomed.